core/editor/src/document/document_message_handler.rs

```diff
@@ -365,6 +366,14 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					responses.push_back(DocumentOperation::SetLayerBlendMode { path: path.clone(), blend_mode }.into());
 				}
 			}
+			SetOpacityForSelectedLayers(opacity) => {
+				let opacity = opacity.clamp(0., 1.);
+				let active_document = self.active_document();
+
+				for path in active_document.layer_data.iter().filter_map(|(path, data)| data.selected.then(|| path)) {
```

Comment:
It would be a little nicer to do `data.selected.then(|| path.clone)` then later `DocumentOperation::SetLayerOpacity { path, opacity }`... if that would work with the borrow checker

Reply:
I think you would then have to chain it with a drain(..). I don't mind the current approach

Reply:
Happy to change it, but I'm not the best person to judge which is best.

Reply:
It compiles as suggested, without requiring `drain`.

Reply:
Could you post the now-working code so @TrueDoctor can give his thoughts?

Reply:
```rust
for path in active_document.layer_data.iter().filter_map(|(path, data)| data.selected.then(|| path.clone())) {
	responses.push_back(DocumentOperation::SetLayerOpacity { path, opacity }.into());
}
```

Reply:
> I think you would then have to chain it with a drain(..). I don't mind the current approach

Ah nevermind, it might just work. when using the clone in the iterator you reduce the code length since you then can get rid of the `path: `